05_concurrency/homework/afterfunc/afterfunc.go

```diff
@@ -0,0 +1,70 @@
+package afterfunc
+
+import (
+	"memcache/interfaces"
+	"sync"
+	"time"
+)
+
+type cache struct {
+	mu      sync.Mutex
+	records map[interfaces.Key]*record
+}
+
+type record struct {
+	value       interface{}
+	ttl         time.Duration
+	deleteTimer *time.Timer
+}
+
+func New() interfaces.Cache {
```

Comment:
Try to avoid returning interfaces. Best practice is to receive interface and return exact type - https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
Hm, ok. I was thinking of several types as the return value type:
* `Cache` (instead of `cache`). Here is the problem - the user can write composite literal `Cache{}`, thus we can meet zero-valued fields (e.g. the map `records` would be `nil`).
* `cache`. It seemed strange to me to return the private struct - the user will not be able to write the type of the value it got.
* `interfaces.Cache`. It seemed most beautiful to me.